src/core/jwk.rs

```diff
@@ -28,6 +28,11 @@ pub struct CoreJsonWebKey {
     #[serde(skip_serializing_if = "Option::is_none")]
     pub(crate) kid: Option<JsonWebKeyId>,
 
+    /// The algorithm intended to be used with this key (https://www.rfc-editor.org/rfc/rfc7517#section-4.4)
+    /// It can either be an algorithm inteded for use with JWS or JWE, or something different.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
```

Comment:
on second thought, let's put this behind the feature flag as well (without `deserialize_option_or_none`), and the existing `VecSkipError`  usage in `JsonWebKeySet` will skip any keys with unrecognized algorithms (when the feature is enabled), which we shouldn't use anyway if this crate doesn't understand them: https://github.com/ramosbugs/openidconnect-rs/blob/1bc4a2f6c741f1c94716c5f4d897b0f3eaa06fbe/src/types.rs#L735

Reply:
yup, we're on the same page

> should make it a non-breaking change that will ignore unrecognized key types

That's what I did at first, but then we couldn't differentiate keys with an unknown alg from keys with no alg specified. I also tried adding an Unknown member to the enum (marked with `serde(other)`), but it made the change very intrusive. So I settled on relying on VecSkipError to skip keys with unknown algs at parsing.

>  on second thought, let's put this behind the feature flag as well (without deserialize_option_or_none), and the existing VecSkipError usage in JsonWebKeySet will skip any keys with unrecognized algorithms

Sounds good! Do you have a preferred name for the feature flag? I understand that you don't want to make a major release just for that, I'm guessing the feature flag would be removed on the next major release?

Reply:
>  I understand that you don't want to make a major release just for that, I'm guessing the feature flag would be removed on the next major release?

yup, hopefully I'll remember to remove it in the next major release. it's probably worth adding a TODO comment next to the feature flag's definition in `Cargo.toml`.

also we can remove the `default` within `serde()` since it's implied by `Option<_>` in the absence of a custom deserializer.